android/src/main/java/com/oblador/keychain/cipherStorage/CipherStorageKeystoreAESCBC.java

```diff
@@ -232,7 +258,16 @@ private String decryptBytes(Key key, byte[] bytes) throws CryptoFailedException
             IvParameterSpec ivParams = new IvParameterSpec(bytes, 0, ivLength);
             cipher.init(Cipher.DECRYPT_MODE, key, ivParams);
 
-            byte[] decryptedBytes = cipher.doFinal(bytes, ivLength, bytes.length - ivLength);
+            // decrypt the bytes using cipher.doFinal(). Using a CipherInputStream for
+            // decryption has historically led to issues
+            // on the Pixel family of devices.
+            // see https://github.com/oblador/react-native-keychain/issues/383
+            byte[] _decryptedBytes = cipher.doFinal(bytes, ivLength, bytes.length - ivLength);
+
+            // removing padding is required to work around a decryption padding issue with
+            // some Pixel devices e.g. for strings of length 256
+            byte[] decryptedBytes = maybeRemovePKCS7Padding(_decryptedBytes, ivLength);
```

Comment:
just checking, this is the only non-whitespace change?

Reply:
yes

Reply:
Posting here for posterity:
- Roughly speaking, `AES/CBC/PKCS7Padding` cipher decrypts certain string lengths as `AES/CBC/NoPadding`, violating PKCS#7 conformance in terms of padding scheme 🤔
- Subscribed to https://issuetracker.google.com/issues/357896946 👀